### PR TITLE
fix: sync the page title regarding the title config

### PR DESCRIPTION
**Pull request by Koooooo-7**
## Summary
Currently, on the v5 changes, each headings/links on the sidebar has the `title` via `#addTextAsTitleAttribute`.
https://github.com/docsifyjs/docsify/blob/42f25482fa1ea6ec666800145f6da33e9119aa8c/src/core/render/index.js#L20

After the `event` refactor, the page `title` update function changed, especially the search page title.

---
Example from our site.
v4:
   On the main content  load the page title such as `Configuration`, no matter you switch the sections to `Configuration`.
   or directly jump to its sub headings [configuration?id=alias](https://docsify.js.org/#/configuration?id=alias).

    The page title is always `Configuration`.

v5 as-is:
   On the main content  load the page title such as `Configuration`, switch the sections to `Configuration`. (same to v4 )
   :warning:  directly jump to its sub headings [configuration?id=alias](https://docsify-preview-git-fork-eldemcan-patch-1-docsifyjs.vercel.app/preview/#/configuration?id=alias) via the URL，it will update the page title to `sub title` (`alias`) instead.
   ⚠️  On the search function, when clicks the search result, it comes to the mess innerText as page title instead. 

The behavior for the sidebar headings for main content, I think it makes sense to indicate the specific target and respect the title.
So, I fix the search problem via a consistent way that all retrieve page title from to the `title` attribute also.
Now, it will use the `${name} - ${title}` as the page title, if no name config, use title directly.

## What kind of change does this PR introduce?

Bugfix
## For any code change,

- [ ] Related documentation has been updated, if needed
- [ ] Related tests have been added or updated, if needed

## Does this PR introduce a breaking change?

Yes
No

No
## Tested in the following browsers:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by trusktr** [comment]
Hmm, I see yeah it is inconsistent depending how you land on the page. Clicking on your link, the title is `alias`, but if you navigate to the page from inside Docsify, the title is `Configuration`.

Also I think that it would be better for the title to be `Docsify - Page` for each page whre `Page` is the name of the page, or similar. F.e. `Docsify - Configuration`, `Docsify - Quick start`, etc.

**Comment by Koooooo-7** [comment]
> Hmm, I see yeah it is inconsistent depending how you land on the page. Clicking on your link, the title is `alias`, but if you navigate to the page from inside Docsify, the title is `Configuration`.
> 
> Also I think that it would be better for the title to be `Docsify - Page` for each page whre `Page` is the name of the page, or similar. F.e. `Docsify - Configuration`, `Docsify - Quick start`, etc.

Yea, It makes sense that we could concat the `name` of the site as a prefix,

File: src/core/event/index.js [hunk 1]
```diff
@@ -296,13 +296,20 @@ export function Events(Base) {
...
-      const currentTitle = dom.find(
-        `.sidebar a[href='${currentPath}']`,
-      )?.innerText;
+      const currentSection = dom
+        .find(`.sidebar a[href='${currentPath}']`)
+        ?.getAttribute('title');
+
+      const currentTitle = name
+        ? currentSection
+          ? `${name} - ${currentSection}`
```
Comment (sy-records) [inline src/core/event/index.js:310]: I suggest putting the site title after the page title.

```suggestion
          ? `${currentSection} - ${name}`
```

**Review by sy-records — DISMISSED** [review]

File: src/core/event/index.js [hunk 2]
```diff
@@ -296,13 +296,20 @@ export function Events(Base) {
     // =========================================================================
     /**
      * Handle rendering UI element updates and new content
+     * @param {string} name the site name
```
Comment (sy-records) [inline src/core/event/index.js:299]:
```suggestion
```

File: src/core/event/index.js [hunk 3]
```diff
@@ -296,13 +296,20 @@ export function Events(Base) {
...
      * @void
      */
-    onRender() {
+    onRender(name) {
       const currentPath = this.router.toURL(this.router.getCurrentPath());
```
Comment (sy-records) [inline src/core/event/index.js:303]: Why not use it like this?

```suggestion
    onRender() {
      const { name } = this.config;
      const currentPath = this.router.toURL(this.router.getCurrentPath());
```

Comment (Koooooo-7) [reply src/core/event/index.js]: update.

File: src/core/event/index.js (hunk 1 above)
Comment (Koooooo-7) [reply src/core/event/index.js:310]: I think the format are both fine, and I test them locally.
Personally I agree with @trusktr  to do something like `docsify - Configuration` to mention the site first.
Cos I checked some of the v4 sites behaviors such as the [docsify-themeable/QuickStart](https://jhildenbiddle.github.io/docsify-themeable/#/quick-start?id=new-site), they looks fine.

Maybe we could have a discuss on this.

Comment (Koooooo-7) [reply src/core/event/index.js:310]: Also, we could introduce a `pageTitleFormatter` config, by default use the `name - title`, users can concat them on what they want in v5+.
The Formatter kinda like:
```js
<script>
  window.$docsify = {
    pageTitleFormatter: (config, section) => { },
  
  };
</script>
```

**Review by sy-records — APPROVED** [review]

File: src/core/event/index.js (hunk 1 above)
Comment (paulhibbitts) [reply src/core/event/index.js:310]: I would also prefer page title first, then site title, as page name is often the more valuable source of contextual info (i.e. looking at open tab titles, list of history, bookmarks etc.).

**Review by paulhibbitts — APPROVED** [review]

**Review by sy-records — APPROVED** [review]

**Comment by BrysonC** [comment]
Since the `name` configuration can "also contain custom HTML for easier customization" (due to its primary/first use in the sidebar) using the name for page title generation can produce some ugly results.

I liked @Koooooo-7 recommendation to make a `pageTitleFormatter` config option.

**Pull request merged**